storybookjs/storybook
# Core: Restore Docs `useParameter` using `DOCS_PREPARED`

Closes #21798 

Telescoping on https://github.com/storybookjs/storybook/pull/22120

## What I did

Added a new event `DOCS_PREPARED` analogous to `STORY_PREPARED` that emits the *parameters* of the current docs entry.

 - That's the component's parameters for CSF docs (autodocs + `.stories.mdx`), and attached MDX docs.
 - That's the project's parameters for unattached MDX docs.

This is in contrast to https://github.com/storybookjs/storybook/pull/22071 which added more impactful events (`PROJECT_PREPARED` and `COMPONENT_PREPARED`) which we should think a little more about.

## How to test

1. Run a sandbox, check toolbar (in particular backgrounds) behaviour
	- Check attached MDX, unattached.
3. Check the same for composition!
4. See tests


## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
